### DISCUSSION TITLE
Creating definitions/lemmas/theorems without id label

### DISCUSSION THREAD

**Discussion by fulem** [quarto-dev/quarto-cli#3771] [Q&A]
How we can avoid having to give an explicit label in the environments of definition, theorems, etc ?

Typically, we don't want to explicitly label ALL environments, except those that are important and will be referenced later.
Suppose for example we have lot of examples that will never be referenced:
```
::: {#exm-a}
A
:::
::: {#exm-b}
B
:::
::: {#exm-c}
C
:::
```
Here i need to keep inventing myself names likes `exm-a`, `exm-b`, `exm-c`, and so on.

A possible solution is to put only the prefix:
```
::: {#exm-}
A
:::
::: {#exm-}
B
:::
::: {#exm-}
C
:::
```
But this is creating html div tags with the same id `exm-`, a bad practice.

Any idea?

**Comment by mcanouil** [comment]
The id is how Quarto "guess" what kind of environment it should apply.

You can always, define/use the classes manually using the full fenced div notation, but this will be more tedious than simply increment your id (simpler with digits).

There is no problem to use the same id for several things, unless you start referencing those elements.

Final note, don't use only the prefix, be explicit even for yourself, e.g., `#exm-nocite`.

**Reply by fulem** [reply]
Hi. Writing 
```
::: {.example} ... :::
```
or
```
::: {.theorem .example} ... :::
```
doesn't work, Quarto needs the id to known it is an example. This is wrong design, if i don't need a labeled environment then i don't need to invent an unique name.  

Compare with bookdown, we write for an unlabeled example
```
::: {.example} ... :::
```
and for a labeled example
```
::: {.example #id} ... :::
```
This is better, don't you think?

**Reply by cscheid** [reply]
This is not a change we're willing to make at this point for a variety of reasons (it will interfere with a number of other assumptions in quarto's crossref subsystem, as one). However, you can write a lua filter yourself that will automatically generate IDs by checking the div classes yourself.
